store/index-refactor.js

```diff
@@ -0,0 +1,23 @@
+// Facilitate Vue reactivity via 'Vue.set' and 'Vue.delete'
+import Vue from "vue";
+
+export const state = () => ({
+    // columnToCategoryMapping: {}
+});
+
+export const getters = {
+
+};
+
+
+export const mutations = {
+    alterColumnCategoryMapping(p_state, activeCategory, columnName) {
```

Comment:
minor point, but I think in the context of the store `activeCategory` isn't that intuitive anymore. I guess `targetCategory` would be clearer?

You could also consider adding a top level docstring to the mutation to summarize the intention of it (link a column to a new category, unless it is already linked - or something like that)

Reply:
Good call. Will do both.